pallets/phala/src/stakepool.rs

```diff
@@ -1348,6 +1508,44 @@ pub mod pallet {
 			}
 		}
 
+		fn has_expired_withdrawal(
+			pool_info: &PoolInfo<T::AccountId, BalanceOf<T>>,
+			now: u64,
+			grace_period: u64,
+			releasing_stake: BalanceOf<T>,
+		) -> bool {
+			debug_assert!(
+				pool_info.free_stake == Zero::zero(),
+				"We really don't want to have free stake and withdraw requests at the same time"
+			);
+			// If we check the pool withdraw_queue here, we don't have to remove a pool from
```

Comment:
Not relevant now?

Reply:
Oh, we should remove `WithdrawalQueuedPools` completely

Reply:
done